local_flow/intent/src/cart_baseline_flow.py

```diff
@@ -123,38 +123,41 @@ def get_model_config(self):
     @enable_decorator(batch(gpu=1, cpu=8, image=os.getenv('BASE_IMAGE')),
                       flag=os.getenv('EN_BATCH'))
     # @ environment decorator used to pass environment variables to Batch instance
-    @environment(vars={'WANDB_API_KEY': os.getenv('WANDB_API_KEY'),
+    @environment(vars={'NEPTUNE_PROJECT': os.getenv('NEPTUNE_PROJECT'),
+                       'NEPTUNE_API_TOKEN': os.getenv('NEPTUNE_API_TOKEN'),
+                       'NEPTUNE_CUSTOM_RUN_ID': os.getenv('NEPTUNE_CUSTOM_RUN_ID'),
+                       'WANDB_API_KEY': os.getenv('WANDB_API_KEY'),
                        'WANDB_ENTITY': os.getenv('WANDB_ENTITY'),
                        'BASE_IMAGE': os.getenv('BASE_IMAGE'),
                        'EN_BATCH': os.getenv('EN_BATCH')})
-    # @ custom pip decorator for pip installation on Batch instance
-    @pip(libraries={'wandb': '0.10.30'})
+    # un-comment if provided image does not contain required packages
+    # @pip(libraries={'wandb': '0.10.30', "neptune-client": "0.13.3", "neptune-tensorflow-keras": "0.9.9"})
     @step
     def train_model(self):
         """
         Train an intention prediction model on GPU.
         """
-        import wandb
+        
         from model import train_lstm_model
+        from utils import ExperimentTracker
 
-        assert os.getenv('WANDB_API_KEY')
-        assert os.getenv('WANDB_ENTITY')
 
-        # initialize wandb for tracking
-        wandb.init(entity=os.getenv('WANDB_ENTITY'),
-                   project="cart-abandonment",
-                   id=current.run_id,
-                   config=self.config,
-                   resume='allow',
-                   reinit=True)
+        # initialize neptune or wandb for tracking
+        tracker = ExperimentTracker(
+            name='neptune', # or 'wandb'
```

Comment:
The argument in the `ExperimentTracker` is `tracker_name` not `name`
```suggestion
            tracker_name='neptune', # or 'wandb'
```

Reply:
Also passing the tracker name as a string makes the pipeline less configurable since the source code has to be modify in order to change the type of experiment tracker. A better approach (IMO) would be to detect the presence of the environment variables defined in the `.env` if the details for `wandb` are found then we pass an `ExperimentTrackerType (Enum)` that points to `wandb`, otherwise we pass `neptune` also as a type.

What do you think?

Reply:
I fully agree with you, I think it's a good idea, let's do it! 

That was my initial approach but I saw the following decorator that got me thinking maybe the user should specify to avoid conflict since it was a bit of a challenge for me to test the code with MetaFlow.
```
@environment(vars={'WANDB_API_KEY': os.getenv('WANDB_API_KEY'),
                       'WANDB_ENTITY': os.getenv('WANDB_ENTITY'),
                       'BASE_IMAGE': os.getenv('BASE_IMAGE'),
                       'EN_BATCH': os.getenv('EN_BATCH')}) 
```

Reply:
>  ...pass an ExperimentTrackerType (Enum)...

Just to be clear, can you give me an example of this `ExperimentTrackerType (Enum)`?

Reply:
I made some changes, let me know what you think!